### PR TITLE
Simplify subtest scoring

### PR DESCRIPTION
**Pull request by gsnedders**

### CODE REVIEW

**Comment by foolip** [comment]
@gsnedders does this change scores at all, or is it a pure refactor?

**Comment by gsnedders** [comment]
> @gsnedders does this change scores at all, or is it a pure refactor?

Plz check, but this should be a pure refactor (hence the lack of any tests).

File: lib/browser-specific.js [hunk 1]
```diff
@@ -194,42 +195,6 @@ function scoreSubtests(browserSubtests) {
...
-              break;
-            }
-          }
-        }
-      }
-    }
-
-    // Move all with the smallest name on.
-    prevName = name;
-    onSameSubtest.forEach(subtest => subtest.moveNext());
```
Comment (gsnedders) [inline lib/browser-specific.js:230]: Looking at this again, I do wonder if we need to change moving the iterator in the first commit. Hopefully the tests give that away?

**Comment by gsnedders** [comment]
> > @gsnedders does this change scores at all, or is it a pure refactor?
> 
> Plz check, but this should be a pure refactor (hence the lack of any tests).

This indeed produces an empty diff.

File: lib/browser-specific.js (hunk 1 above)
Comment (gsnedders) [reply lib/browser-specific.js:230]: No, because we've continued in the case where we have fewer results.

File: lib/browser-specific.js [hunk 2]
```diff
@@ -126,42 +126,34 @@ function scoreSubtests(browserSubtests) {
...
-        denominator += 1;
-        if (onSameSubtest.length == browserSubtests.length - 1) {
-          if (onSameSubtest.every(s => TEST_PASS_STATUSES.includes(
+    if (name === prevName) {
+      // There actually (rarely) exist distinct subtests with the same name in
+      // the data, usually because of unprintable characters. This can influence
+      // the result as we may mismatch results (i.e. if some browser has results
+      // for one duplicate-named subtest but not another).
+      //
+      // We avoid scoring another subtest result as otherwise we end up with the
```
Comment (foolip) [inline lib/browser-specific.js:135]: Great comments!

**Review by foolip — APPROVED** [review]

**Pull request merged**